Sources/Workspace/Workspace.swift

```diff
@@ -1338,48 +1334,59 @@ extension Workspace {
-            return node.item.0.dependenciesRequired(for: node.item.1).compactMap({ dependency in
+        // continue to load the rest of the manifest for this graph
+        let allManifestsWithPossibleDuplicates = try topologicalSort(inputManifests.map{ KeyedPair($0, key: URLAndFilter(url: $0.url, filter: .everything)) }) { node in
+            return node.item.dependenciesRequired(for: node.key.filter).compactMap{ dependency in
                 let url = config.mirrors.effectiveURL(forURL: dependency.url)
                 // FIXME: this should not block
+                // note: loadManifest emits diagnostics in case it fails
                 let manifest = loadedManifests[url] ?? temp_await { self.loadManifest(forURL: url, diagnostics: diagnostics, completion: $0) }
                 loadedManifests[url] = manifest
-                return manifest.flatMap({ KeyedPair(($0, dependency.productFilter), key: NameAndFilter(name: $0.name, filter: dependency.productFilter)) })
-            })
+                return manifest.flatMap({ KeyedPair($0, key: URLAndFilter(url: $0.url, filter: dependency.productFilter)) })
+            }
         }
 
-        var deduplication: Set<String> = []
-        var allManifests: [KeyedPair<(Manifest, ProductFilter), NameAndFilter>] = []
+        // remove duplicates of the same manifest
+        // FIXME: this drops duplicates with different filters but filter is not in use in this case. we should clean this API up
+        var deduplication: Set<PackageIdentity> = []
+        var allManifests = [(manifest: Manifest, productFilter: ProductFilter)]()
         for node in allManifestsWithPossibleDuplicates {
-            if deduplication.contains(node.item.0.name) {
+            let identity = PackageIdentity(url: node.item.url)
+            if deduplication.contains(identity) {
                 continue // A duplicate.
             } else {
-                allManifests.append(node)
-                deduplication.insert(node.item.0.name)
+                allManifests.append((node.item, node.key.filter))
+                deduplication.insert(identity)
             }
         }
 
-        let allDependencyManifests = allManifests.map{ $0.item }.filter{ !root.manifests.contains($0.0) }
-        let deps = try allDependencyManifests.map{ manifest, productFilter -> (Manifest, ManagedDependency, ProductFilter) in
-            guard let dependency = state.dependencies[forURL: manifest.url] else {
+        let dependencyManifests = allManifests.filter{ !root.manifests.contains($0.manifest) }
+
+        // check for overrides attempts with same name but different path
+        let rootManifestsByName = root.manifests.spm_createDictionary{ ($0.name, $0) }
+        dependencyManifests.forEach { manifest, _ in
+            if let override = rootManifestsByName[manifest.name], override.url != manifest.url  {
```

Comment:
I thought that overriding was case-insensitive, but I see that this code was just moved from somewhere else, so it's safest to keep as-is.  But still somewhat surprising, I think.  @aciidb0mb3r do you know why this would be a case-sensitive check?

Reply:
I tried to keep changes minimal in this PR, agreed that this seems wrong